### PR TITLE
IMP: ShipIT changes

### PR DESCRIPTION
**Pull request by poespas**
## Added features:
- Adds config field for "own_packages" to indicate local packges, filtered by a string in package name
- Gets all collaborators for a git package, gotten dynamically from the `git log`
- Adds a checkmark to the frontend to filter on own packages only
- Adds Mooore-icon to own packages
- Adds a "Last updated at" field to the frontend to show the last update time of the package.

### CODE REVIEW

File: src/Builder/WebBuilder.php [hunk 1]
```diff
@@ -20,6 +20,7 @@
 use Twig\Extension\DebugExtension;
 use Twig\Extra\Html\HtmlExtension;
 use Twig\Loader\FilesystemLoader;
+use Composer\Satis\Git;
```
Comment (tdgroot) [inline src/Builder/WebBuilder.php]: Imports need to be sorted.

File: src/Git.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Composer\Satis;
+
+use Exception;
+
+class Git {
+    private $location = null;
+
+    public function __construct() {
```
Comment (tdgroot) [inline src/Git.php]: Remove empty constructor

File: src/Git.php [hunk 3]
```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Composer\Satis;
+
+use Exception;
+
+class Git {
+    private $location = null;
```
Comment (tdgroot) [inline src/Git.php]:
```suggestion
    private ?string $location = null;
```

File: src/Git.php [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    public function setDir(string $location): Git {
+        $this->location = $location;
+
+        return $this;
+    }
+
+    public function getCollaborators() {
+        $resp = $this->runGitCommand("log --format='{\"name\": \"%aN\", \"email\": \"%aE\"}'  | sort |  uniq -c | sort -nr");
```
Comment (tdgroot) [inline src/Git.php:21]:
```suggestion
        if ($this->location === null) {
            return null;
        }

        $resp = $this->runGitCommand("log --format='{\"name\": \"%aN\", \"email\": \"%aE\"}'  | sort |  uniq -c | sort -nr");
```

File: src/Git.php [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        );
+
+        return $colabs;
+    }
+
+    public function runGitCommand(string $command) {
+        $cmds = [];
+
+        $cmds[] = "cd $this->location";
+        $cmds[] = "git --git-dir . --work-tree . $command";
```
Comment (tdgroot) [inline src/Git.php:48]: If you specifiy the `git-dir`, why first cd into that directory?

File: views/assets/js/App/PackageFilter.js [hunk 6]
```diff
@@ -71,13 +74,27 @@ class PackageFilter {
...
         : elem.textContent;
 
+      // Get the "own_packages_only" field
+      let is_own = true;
+
+      if (own_packages_only) {
+        is_own = false;
+
+        window.config.own_packages.forEach((packageName) => {
+          if (filterableContent.toLowerCase().indexOf(packageName) !== -1) {
```
Comment (tdgroot) [inline views/assets/js/App/PackageFilter.js]: Wouldn't it make more sense to explode the package name by `/`, take the first part (the vendor part) and check if that value matches exactly?

**Review by tdgroot — CHANGES_REQUESTED** [review]
Nice work! A few changes requested!

File: src/Git.php (hunk 5 above)
Comment (poespas) [reply src/Git.php:48]: Only the actual `.git` folders are available within the composer cache, not the actual project folders, so adding both `git-dir` and `work-tree` to the `.git` folder would fix this issue, if I were to remove these, it simply would throw out a repository couldn't be found.

File: views/assets/js/App/PackageFilter.js (hunk 6 above)
Comment (poespas) [reply views/assets/js/App/PackageFilter.js]: It does, it's added now.

File: src/Git.php (hunk 4 above)
Comment (poespas) [reply src/Git.php:21]: Added!

File: src/Git.php (hunk 3 above)
Comment (poespas) [reply src/Git.php]: Fixed!

File: src/Builder/WebBuilder.php (hunk 1 above)
Comment (poespas) [reply src/Builder/WebBuilder.php]: Done!

File: src/Git.php (hunk 2 above)
Comment (poespas) [reply src/Git.php]: Done!

**Review by tdgroot — APPROVED** [review]

**Pull request merged**